docs/core/DelegationManager.md

```diff
@@ -0,0 +1,195 @@
+* If the Operator calls this method, they need to submit only the `stakerSignatureAndExpiry`
+* If the Operator's `delegationApprover` calls this method, they need to submit only the `stakerSignatureAndExpiry`
+
+**Effects**: See `delegateTo` above.
+
+**Requirements**: See `delegateTo` above. Additionally:
+* If caller is either the Operator's `delegationApprover` or the Operator, the `approverSignatureAndExpiry` and `approverSalt` can be empty
+* `stakerSignatureAndExpiry` MUST be a valid, unexpired signature over the correct hash and nonce
+
+*Unimplemented as of M2*:
+* `require(!slasher.isFrozen(operator))` is currently a no-op
+
+### Other
+
+#### `undelegate`
+
+```solidity
+function undelegate(address staker) external returns (bytes32 withdrawalRoot)
+```
+
+This method undelegates a Staker from an Operator, decreasing the Operator's shares in the strategies held by the Staker. This can be called by a Staker to undelegate themselves, or by a Staker's delegated Operator (or that Operator's `delegationApprover`).
+
+If the Staker has active shares in the `EigenPodManager`, the Staker is placed into "undelegation limbo."
+
+If the Staker has active shares in any strategy in the `StrategyManager`, this initiates a withdrawal of the Staker's shares.
+
+**Effects**: 
+* `eigenPodManager.forceIntoUndelegationLimbo`: If the Staker has shares in the `EigenPodManager`, they are placed into undelegation limbo and the shares are decremented from the Operator's beacon chain ETH shares.
+* `strategyManager.forceTotalWithdrawal`: If the Staker has shares in any strategy, this method initiates a withdrawal of all shares via the `StrategyManager` withdrawal queue. Each strategy's shares are also decremented from the Operator's shares.
+* If the Staker was delegated to an Operator, this function undelegates them.
+
+**Requirements**:
+* Staker MUST exist and be delegated to someone
+* Staker MUST NOT be an Operator
+* Caller must be either the Staker, their Operator, or their Operator's `delegationApprover`
+* Pause status MUST NOT be set: `PAUSED_UNDELEGATION`
+* `EigenPodManager`: see `forceIntoUndelegationLimbo`
+* `StrategyManager`: see `forceTotalWithdrawal`
+
+#### `increaseDelegatedShares`
+
+```solidity
+function increaseDelegatedShares(address staker, IStrategy strategy, uint256 shares)
+    external
+    onlyStrategyManagerOrEigenPodManager
+```
+
+Called by either the `StrategyManager` or `EigenPodManager` when a Staker's shares for one or more strategies increase. This method is called to ensure that if the Staker is delegated to an Operator, that Operator's share count reflects the increase.
+
+**Entry Points**: This method may be called as a result of the following top-level function calls:
+* `StrategyManager.depositIntoStrategy`
+* `StrategyManager.depositIntoStrategyWithSignature`
+* `EigenPodManager.exitUndelegationLimbo`
+* `EigenPod.verifyWithdrawalCredentials`
+* `EigenPod.verifyBalanceUpdate`
+* `EigenPod.verifyAndProcessWithdrawals`
```

Comment:
idk if you want to add an asterisk or something that explains about these functions not being called when 'undelegation limbo' is active ? the wording of "_may_ be called" seems sufficient

Reply:
i'm gonna leave it as is - just want to give someone an idea of where to look and not complicate it too much. thanks for the note :)